File: AdapterViewFlipperApp/app/src/main/java/com/example/adapterviewflipperapp/MainActivity.java
package com.example.adapterviewflipperapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.AdapterViewFlipper;
import android.widget.ListView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    AdapterViewFlipper adapterViewFlipper;

    CustomAdapter customAdapter;
    ArrayList<ItemsModel> arrayList=new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        adapterViewFlipper = (AdapterViewFlipper) findViewById(R.id.adapterviewflipper);



      arrayList.add(new ItemsModel(R.drawable.afghanisthan,"Afganistan"));
      arrayList.add(new ItemsModel(R.drawable.americaflag,"America"));
      arrayList.add(new ItemsModel(R.drawable.bangladesh,"Bangladesh"));
      arrayList.add(new ItemsModel(R.drawable.bruini,"Bruini"));
      arrayList.add(new ItemsModel(R.drawable.canada,"Canada"));
      arrayList.add(new ItemsModel(R.drawable.china,"China"));
      arrayList.add(new ItemsModel(R.drawable.indiaflag,"India"));
      arrayList.add(new ItemsModel(R.drawable.egypt,"Egypt"));
      arrayList.add(new ItemsModel(R.drawable.japan,"Japan"));
      customAdapter=new CustomAdapter(this,arrayList);
   adapterViewFlipper.setAdapter(customAdapter);
     adapterViewFlipper.startFlipping();
     adapterViewFlipper.setFlipInterval(3000);


    }
}

File: SampleAppLifecycle/app/src/main/java/com/example/sampleapplifecycle/MainActivity.kt
package com.example.sampleapplifecycle

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button

class MainActivity : AppCompatActivity() {
    val TAG ="MainActivityLog"
    lateinit var button:Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(TAG,"onCreate")
        setContentView(R.layout.activity_main)
        button=findViewById(R.id.button)
       button.setOnClickListener {
           startActivity(Intent(this,SecoundActivity::class.java) )

       }
    }

    override fun onStart() {
        Log.d(TAG,"onStart")
        super.onStart()
    }

    override fun onResume() {
        Log.d(TAG,"onResume")
        super.onResume()
        askPermission()
    }
    override fun onPause() {
        Log.d(TAG,"onStart")
        super.onPause()
    }
    override fun onStop() {
        Log.d(TAG,"onStart")
        super.onStop()
    }

    override fun onDestroy() {
        Log.d(TAG,"onDestroy")
        super.onDestroy()
    }
    fun askPermission(){
        Log.d(TAG,"askPermission called")
    }

}

